sklearnex/tests/test_n_jobs_support.py

```diff
@@ -14,18 +14,76 @@
 # limitations under the License.
 # ==============================================================================
 
+import inspect
+import logging
+from io import StringIO
+from multiprocessing import cpu_count
+
 import pytest
+from sklearn.base import BaseEstimator
+from sklearn.datasets import make_classification
+
+from sklearnex.dispatcher import get_patch_map
+from sklearnex.svm import SVC, NuSVC
+
+ESTIMATORS = set(
+    filter(
+        lambda x: inspect.isclass(x) and issubclass(x, BaseEstimator),
+        [value[0][0][2] for value in get_patch_map().values()],
+    )
+)
+
+X, Y = make_classification(n_samples=40, n_features=4, random_state=42)
+
+
+@pytest.mark.parametrize("estimator_class", ESTIMATORS)
+@pytest.mark.parametrize("n_jobs", [None, -1, 1, 2])
+def test_n_jobs_support(estimator_class, n_jobs):
+    def check_estimator_doc(estimator):
+        if estimator.__doc__ is not None:
+            assert "n_jobs" in estimator.__doc__
+
+    def get_sklearnex_logger_stream(level="DEBUG"):
```

Comment:
Did you look into https://docs.pytest.org/en/latest/how-to/logging.html#caplog-fixture

Is there anything special about our logs that we have to reinvent the wheel?

Reply:
By the way, why are we testing logs in the first place? Is this an important message that must be shown to the user. Or is it rather a proxy to check if some statements were executed? If it is the latter, we should maybe do this test differently.

Reply:
Updated to caplog usage (https://github.com/intel/scikit-learn-intelex/pull/1622/commits/4b53223206733813e1924e1a5e0e3bc6b2ad5a14).

Reply:
These messages are important for debugging.